packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -133,7 +134,13 @@ class EditableText extends StatefulWidget {
     this.keyboardType,
     this.onChanged,
     this.onSubmitted,
-  }) : super(key: key) {
+    List<TextInputFormatter> inputFormatters,
+  }) : inputFormatters = maxLines == 1 
+           ? ((inputFormatters ?? <TextInputFormatter>[])
+                 ..add(BlacklistingTextInputFormatter.singleLineFormatter)
```

Comment:
mutating the list is sketchy. if someone creates one then passes it in each time, it'll get progressively longer.
if someone passes in a const list, it'll crash.